pkg/metrics/serve.go

```diff
@@ -75,39 +81,45 @@ func NewServer(ctx context.Context, opts ...ServerOpt) (*Server, error) {
 	return &s, nil
 }
 
-func (s *Server) CollectDaemonMetrics(ctx context.Context) error {
+func (s *Server) CollectMetrics(ctx context.Context) {
+	// Collect FS metrics from daemons.
+	daemons := s.pm.ListDaemons()
+	for _, d := range daemons {
+		for _, i := range d.Instances.List() {
+			var sid string
+
+			if i.GetMountpoint() == d.HostMountpoint() {
+				sid = ""
+			} else {
+				sid = i.SnapshotID
+			}
+
+			fsMetrics, err := d.GetFsMetrics(sid)
+			if err != nil {
+				log.G(ctx).Errorf("failed to get fs metric: %v", err)
+				continue
+			}
+			s.fsCollector.Metrics = fsMetrics
+			s.fsCollector.ImageRef = i.ImageID
+
+			s.fsCollector.Collect()
+		}
+
+	}
+
+	// Collect snapshotter metrics.
+	s.snCollector.Collect()
```

Comment:
This is where collecting nydusd io metrics. Snapshotter should by default has metrics

Reply:
Yes, I will fix this.